e2e/pages/events/manageLegalCounsellorsEvent.page.js

```diff
@@ -0,0 +1,45 @@
+const {I} = inject();
+
+module.exports = {
+  fields: function (index) {
+
+    return {
+      legalCounsellor: {
+        firstName: `#legalCounsellors_${index}_firstName`,
+        lastName: `#legalCounsellors_${index}_lastName`,
+        organisationGroup: `#legalCounsellors_${index}_${index}`,
+        email: `#legalCounsellors_${index}_email`,
+        telephone: `[id="legalCounsellors_${index}_telephoneNumber"]`,
+      },
+    };
+  },
+
+  async addLegalCounsellor(legalRepresentative) {
+
+    const elementIndex = await this.getActiveElementIndex();
+
+    if(legalRepresentative.firstName) {
+      I.fillField(this.fields(elementIndex).legalCounsellor.firstName, legalRepresentative.firstName);
+    }
+    if(legalRepresentative.lastName) {
+      I.fillField(this.fields(elementIndex).legalCounsellor.lastName, legalRepresentative.lastName);
+    }
+    if(legalRepresentative.email) {
+      I.fillField(this.fields(elementIndex).legalCounsellor.email, legalRepresentative.email);
+    }
+    if(legalRepresentative.telephone) {
+      I.fillField(this.fields(elementIndex).legalCounsellor.telephone, legalRepresentative.telephone);
+    }
+    if(legalRepresentative.organisation) {
+      await within(this.fields(elementIndex).legalCounsellor.organisationGroup, async () => {
+        I.fillField('#search-org-text', legalRepresentative.organisation);
```

Comment:
We make use of this component in a couple of tests now, do you think there would be any advantage in making it into a fragment so that we can nicely reuse the functionality?

Reply:
Done!